data-hibernate-jpa/src/test/java/io/micronaut/data/hibernate/querygroupby/MicronautTaskRepository.java

```diff
@@ -0,0 +1,16 @@
+package io.micronaut.data.hibernate.querygroupby;
+
+import io.micronaut.data.annotation.Query;
+import io.micronaut.data.annotation.Repository;
+import io.micronaut.data.hibernate.entities.MicronautTask;
+import io.micronaut.data.repository.CrudRepository;
+
+import java.util.List;
+@Repository
+public interface MicronautTaskRepository extends CrudRepository<MicronautTask, Long> {
+    @Query("select count(*), year(t.dueDate) from MicronautTask t group by year(t.dueDate)")
```

Comment:
This works when written as 
```
@Query("select count(*) as number, year(t.dueDate) as year from MicronautTask t group by year(t.dueDate)")
```
because of this line of our code in AbstractHibernateRepository 
```
Set<String> properties = tuple.getElements().stream().map(TupleElement::getAlias).collect(Collectors.toCollection(() -> new TreeSet<>(String.CASE_INSENSITIVE_ORDER)));
```
expects that TupleElement::getAlias is not null and is one of properties Set which is number, year in our case. Then query can be executed.